spec/requests/basic_auth_spec.rb

```diff
@@ -35,60 +28,134 @@
+
+    it 'posting to the create vacancy endpoint requires basic auth' do
+      school = create(:school)
+
+      path = school_vacancies_path(school.id)
+      post path, params: { vacancy: { foo: :bar } }
+
+      expect(response.status).to eq(401)
+    end
+
+    context 'and valid global basic auth has been provided' do
+      let(:username) { 'username' }
+      let(:password) { 'foobar' }
+      let(:encoded_credentials) do
+        ActionController::HttpAuthentication::Basic.encode_credentials(username, password)
+      end
+
+      before(:each) do
+        fake_env = double.as_null_object
+        allow(Figaro).to receive(:env).and_return(fake_env)
+        allow(fake_env).to receive(:http_user).and_return(username)
+        allow(fake_env).to receive(:http_pass).and_return(password)
+      end
+
+      it_behaves_like 'basic auth is required', '/schools', :hiring_staff_http_user, :hiring_staff_http_pass
+
+      it 'posting to the create vacancy endpoint still requires the hiring staff basic auth' do
+        school = create(:school)
 
-    it_behaves_like 'requires basic auth'
+        path = school_vacancies_path(school.id)
+        post path, params: { vacancy: { foo: :bar } }, env: { 'HTTP_AUTHORIZATION': encoded_credentials }
+
+        expect(response.status).to eq(401)
+      end
+    end
   end
 
   context 'when in production' do
-    before(:each) { stub_env_based_authentication }
+    before(:each) do
+      stub_authenticate(return_value: true)
+      stub_authenticate_hiring_staff(return_value: true)
+    end
+
+    it_behaves_like 'basic auth is required', '/', :http_user, :http_pass
+
+    it 'posting to the create vacancy endpoint requires basic auth' do
+      school = create(:school)
+
+      path = school_vacancies_path(school.id)
+      post path, params: { vacancy: { foo: :bar } }
+
+      expect(response.status).to eq(401)
+    end
 
-    it_behaves_like 'requires basic auth'
+    context 'and valid global basic auth has been provided' do
+      let(:username) { 'username' }
+      let(:password) { 'foobar' }
+      let(:encoded_credentials) do
+        ActionController::HttpAuthentication::Basic.encode_credentials(username, password)
+      end
+
+      before(:each) do
+        fake_env = double.as_null_object
+        allow(Figaro).to receive(:env).and_return(fake_env)
+        allow(fake_env).to receive(:http_user).and_return(username)
+        allow(fake_env).to receive(:http_pass).and_return(password)
+      end
+
+      it_behaves_like 'basic auth is required', '/schools', :hiring_staff_http_user, :hiring_staff_http_pass
+
+      it 'posting to the create vacancy endpoint still requires the hiring staff basic auth' do
+        school = create(:school)
+
+        path = school_vacancies_path(school.id)
+        post path, params: { vacancy: { foo: :bar } }, env: { 'HTTP_AUTHORIZATION': encoded_credentials }
+
+        expect(response.status).to eq(401)
```

Comment:
can we change it to `expect(response).to have_http_status(:unauthorized)` as it's easier to read